### PR TITLE
Update SimpChinese installer and qt translation

### PR DESCRIPTION
**Pull request by deluxghost**
the qt translation template is downloaded from qt website (qt 5.12)
i'm not sure if this is ok
mention me if i should change

### CODE REVIEW

**Comment by deluxghost** [comment]
is there any dev looked this pr ever?

File: installer/Translations/SimpChinese.wxl [hunk 1]
```diff
@@ -2,48 +2,49 @@
 <WixLocalization Culture="zh-cn" xmlns="http://schemas.microsoft.com/wix/2006/localization">
   <String Id="LANG">2052</String>
 
+  <!-- By deluxghost -->
```
Comment (davidebeatrici) [inline installer/Translations/SimpChinese.wxl]: I would remove this line, as we have a dedicated file (`AUTHORS`) for the contributors.

**Comment by davidebeatrici** [comment]
`qt_zh_CN.ts` seems to be already present in Qt 5: https://code.qt.io/cgit/qt/qttranslations.git/tree/translations/qt_zh_CN.ts

**Comment by deluxghost** [comment]
> `qt_zh_CN.ts` seems to be already present in Qt 5: https://code.qt.io/cgit/qt/qttranslations.git/tree/translations/qt_zh_CN.ts

after quick ctrl-f i find some mistakes/missing translations in the official file, looks like it's not updated for years?

i wonder should i contribute to qt first, or just update these strings for mumble in this pr?

**Comment by davidebeatrici** [comment]
> after quick ctrl-f i find some mistakes/missing translations in the official file, looks like it's not updated for years?

https://code.qt.io/cgit/qt/qttranslations.git/log/translations/qt_zh_CN.ts

4 commits in total, only small changes since the file has been imported in 2011.

> i wonder should i contribute to qt first, or just update these strings for mumble in this pr?

I would update the strings in this pull request first, so that we have the updated translation for all versions of Qt 5.

**Comment by deluxghost** [comment]
will try to update and merge these changes recently

**Comment by davidebeatrici** [comment]
No problem.

Please enable Git's auto-rebase feature by executing `git config --global pull.rebase true`, that way merge commits are not created when you pull from a remote branch.

**Comment by Krzmbrzl** [comment]
To me it appears as if the commits in this PR only have to be squashed and then it should be ready to be merged...

**Comment by deluxghost** [comment]
sorry for the delay. I was busy these months. I will update this pr before February (I have spring festival to do this)

**Comment by Krzmbrzl** [comment]
No problem :)
With your permission I think the squashing can even be done by us (or more specifically @davidebeatrici  as I don't appear to have the needed permissions) ^^

**Comment by deluxghost** [comment]
thanks, however I just need to re-check the content to make sure nothing goes wrong

**Comment by Krzmbrzl** [comment]
Alright then :+1:

**Comment by deluxghost** [comment]
ummm... looking good?

**Comment by Krzmbrzl** [comment]
`qt_zh_CN.ts` is a new file, correct? Normally these have to be added to a `.pri` file in order for Qt to find and use them, but given that this is the translation for some qt stuff directly (that's what it is, isn't it?), it might not be necessary.

I assume you have already verified that the translation will actually show up in the proper places?

**Comment by deluxghost** [comment]
Not yet. I spent one whole day to try to setup a building environment but failed. Looks like we don't need to add the file name somewhere, but I'm not sure the translation will be actually applied.

**Comment by Krzmbrzl** [comment]
I assume you are using windows, right? It really is a pain to set the system up on windows and to be honest I haven't done it myself yet.
I assume you have found https://wiki.mumble.info/wiki/Building Windows already?

Do you have a specific problem with building mumble? If so, maybe we can help, if you share it with us :)
Also it might be good to join the IRC channel via matrix (https://riot.im/app/#/room/#mumble:matrix.org) for this kind of stuff.

**Comment by deluxghost** [comment]
looks like the `qttranslations` folder is entirely ignored currently, i tried to switch language to `nl` and `tr`, which have files in that folder (and zh_CN in my build), no translation for buttons, both official and my private build

i dont know c++ and qt, so i have no workaround for this issue :(

**Comment by Krzmbrzl** [comment]
Have a look at https://github.com/mumble-voip/mumble/commit/e87028aea99fcbaed0e4cc004050dee83bafb47b (only the files `mumble.pro` and `mumble_qt.qrc` are relevant I think). From there I suspect that you have to create an alias for the file and also add them to the translations in the pro-file...

**Comment by Krzmbrzl** [comment]
Oh wait a sec. It seems something has changed in that regard until then...
https://github.com/mumble-voip/mumble/blob/72e1f47b7b13d2ffd2a98cb2fd79c10674376698/src/mumble/mumble.pro#L709-722

**Comment by davidebeatrici** [comment]
Ideally we should have granular control, perhaps by adding a configuration file to the directory.

That would allows us to only override certain translations.

**Comment by Krzmbrzl** [comment]
Okay I extended the script accordingly in #3955. It's untested though, but in theory this should give us what we need :crossed_fingers: 

In theory it works like so:
There's a new file called `translations.conf` in the `qttranslations` directory. In there you'll have to add a line containing `overwrite qt_zh_CN.ts`. Then save and build again.
In theory that should do it.

You can verify that it worked by opening `mumble_qt_auto.qrc` (probably located in the `mumble` directory or in the root dir) and in there, there should be a line `<file alias="qt_zh_CN.qm">PATH</file>\n` where `PATH` should point to your translation file (or rather the `*.qm` file Qt has generated from it.

Do you know how to merge the linked PR into your branch to test this out?
(I'm sorry for not being able to test it myself, but I don't have my dev-laptop at hand right now)

**Comment by deluxghost** [comment]
i merged #3955 into my branch and fixed the python script, now the .qrc file looks good.

but to make translations actually show, someone might wanna change something here:

https://github.com/mumble-voip/mumble/blob/72e1f47b7b13d2ffd2a98cb2fd79c10674376698/src/mumble/main.cpp#L399-L408

**Comment by deluxghost** [comment]
`nl` and `tr` locale might have the similar issue, the official translation is uncompleted, the fallback version is better. idk if the translators for these languages want `overwrite` too

**Comment by Krzmbrzl** [comment]
> and fixed the python script

Thanks - apparently my Python-Fu wasn't strong with this one ^^

> nl and tr locale might have the similar issue, the official translation is uncompleted, the fallback version is better.

Do you speak these languages as well? Or did you simply compare which translation seems to translate more strings? :thinking: 

> but to make translations actually show, someone might wanna change something here

Indeed. I changed the linked PR and force-pushed it's changes to your branch. Hopefully it takes effect now :)
Remember that you have to force-pull from your branch in order for these changes to be reflected locally on your machine, before trying to build again.

**Comment by deluxghost** [comment]
> Do you speak these languages as well? Or did you simply compare which translation seems to translate more strings?

i dont speak these languages, just a quick compare. the related translator should check themselves.

i added missing overwrite config, and built mumble on macos (except i must ignore some deprecated warnings because my newer local qt 5.14), and victory! the translation works now

**Comment by Krzmbrzl** [comment]
> i added missing overwrite config

Ah right I wanted to add that back in xDD

> except i must ignore some deprecated warnings because my newer local qt 5.14

What kind of warnings? Did they occur before as well? :thinking: 

> the translation works now

Perfect! :+1:

**Comment by deluxghost** [comment]
an example of the warning:

```
../HostAddress.cpp:142:7: error: 'sprintf' is deprecated: Use asprintf(), arg() or QTextStream instead [-Werror,-Wdeprecated-declarations]
                        qs.sprintf("[%x:%x:%x:%x:%x:%x:%x:%x]", ntohs(shorts[0]), ntohs(shorts[1]), ntohs(shorts[2]), ntohs(shorts[3]), ntohs(shorts[4]), ntohs(shorts[5]), ntohs(shorts[6]), ntohs(...
                           ^
/usr/local/Cellar/qt/5.14.0/lib/QtCore.framework/Headers/qstring.h:381:5: note: 'sprintf' has been explicitly marked deprecated here
    QT_DEPRECATED_X("Use asprintf(), arg() or QTextStream instead")
    ^
/usr/local/Cellar/qt/5.14.0/lib/QtCore.framework/Headers/qglobal.h:294:33: note: expanded from macro 'QT_DEPRECATED_X'
#  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
                                ^
/usr/local/Cellar/qt/5.14.0/lib/QtCore.framework/Headers/qcompilerdetection.h:643:55: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#    define Q_DECL_DEPRECATED_X(text) __attribute__ ((__deprecated__(text)))
                                                      ^
1 error generated.
make[2]: *** [../../release/.obj/mumble/HostAddress.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

logs are too long to find others, but they should be unrelated to translation.

i used brew to install the buildenv, same as your macos ci, maybe the ci has same issue

uhhhh check out this: https://dev.azure.com/Mumble-VoIP/Mumble/_build/results?buildId=1162&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=2dac33d1-b0ca-5feb-3c36-18b56319df50

turns out that job built an empty Mumble.app as the artifact

**Comment by Krzmbrzl** [comment]
> they should be unrelated to translation.

Yeah, I agree

> uhhhh check out this: https://dev.azure.com/Mumble-VoIP/Mumble/_build/results?buildId=1162&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=2dac33d1-b0ca-5feb-3c36-18b56319df50
>
> turns out that job built an empty Mumble.app as the artifact

Oh lol. Apparently the macOS CI always succeeds even though the build has failed. That's not how it should be. Thanks for pointing this out! :+1:


Anyways I think as far as this PR goes, it is ready to be merged. I will squash that last commit and then I'll merge it.
Thank you very much for your contribution and patience with the build-process :)

**Comment by davidebeatrici** [comment]
@Krzmbrzl Thank you for pushing this forward.

I see you pulled from `master` without rebasing (d7a6c3c2cd58aac8e8cc7fe5d9eab14be974b76b was created), I would enable `autorebase`:

```
git config --global pull.rebase true
```

**Comment by Krzmbrzl** [comment]
I have it enabled, but this merge was created by resolving the merge conflict via GitHub. Moral of the story: Don't use GitHub UI for these things xD

**Pull request merged**